website/docs/developer/research/research-AuroraServerless.md

```diff
@@ -0,0 +1,96 @@
+    May be multiple timeouts in play.
+  - Scaling up after this point did not cause query performance degradation, despite what was documented. However, the UI is poorly made, and may be reporting incorrectly.
+- Connections canceled due to scaling raise an error in SQLAlchemy.
+  If the query is ongoing:
+  ```
+  [SQL: 
+  SELECT pg_sleep(600);
+  ]
+  (Background on this error at: http://sqlalche.me/e/14/e3q8)
+  Traceback (most recent call last):
+    File "...\venv\lib\site-packages\sqlalchemy\engine\base.py", line 1770, in _execute_context
+      self.dialect.do_execute(
+    File "...\venv\lib\site-packages\sqlalchemy\engine\default.py", line 717, in do_execute
+      cursor.execute(statement, parameters)
+  psycopg2.errors.AdminShutdown: terminating connection due to serverless scale event timeout
+  SSL connection has been closed unexpectedly
+  
+  
+  The above exception was the direct cause of the following exception:
+  
+  Traceback (most recent call last):
+    File ".../main.py", line 85, in <module>
+      connection.execute(wait_command)
+      ...
+  sqlalchemy.exc.OperationalError: (psycopg2.errors.AdminShutdown) terminating connection due to serverless scale event timeout
+  SSL connection has been closed unexpectedly
+  ```
+  
+  If the query is starting up:
+  ```
+  Database connection failed due to (psycopg2.OperationalError) FATAL:  terminating connection because backend initialization completed past serverless scale point
+  
+  (Background on this error at: https://sqlalche.me/e/14/e3q8)
+  Traceback (most recent call last):
+    ...
+    File "...\venv\lib\site-packages\psycopg2\__init__.py", line 122, in connect
+      conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
+  psycopg2.OperationalError: FATAL:  terminating connection because backend initialization completed past serverless scale point
+  
+  
+  The above exception was the direct cause of the following exception:
+  
+  Traceback (most recent call last):
+    File ".../main.py", line 84, in <module>
+      with engine.begin() as connection:
+    File "...\engine.py", line 393, in begin
+      conn = self.connect()
+    ...
+  sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) FATAL:  terminating connection because backend initialization completed past serverless scale point
+  
+  (Background on this error at: https://sqlalche.me/e/14/e3q8)
+  ```
+  
+  - Retrying when these errors occur should be sufficient to sidestep scaling errors, as scaling does not take long.
+  :::warning
+    Disconnections caused by these errors are NOT reported to Cloudwatch. Connecting lambdas/modules should have reliable and robust error logging.
+  :::
+
+### Future Direction
+- Modifications should be made to database code to account for AdminShutdown, OperationalError, and connection timeout issues.
```

Comment:
Did you want to speak to performing a potential check query like Cumulus does or another probe query to wake things up or is managing errors and timeouts with retries sufficient enough to overcome any errors or issues?

Reply:
I'd prefer the later. Their strategy of a test query works for checking for spin-up, but doesn't cover errors caused by scaling, so we'd need error-handling logic at each step anyway. It also doubles time spent waiting on network latency, when we would just send our actual query and be done with it.